### ISSUE TITLE
src/metric/auc.cc:322: Check failed: auc <= local_area

### ISSUE THREAD

**Issue by zhuwang46** [dmlc/xgboost#10183]
The error message occurred after changing nrounds from 32 to 33:
[weight.txt](https://github.com/dmlc/xgboost/files/14971512/weight.txt)

<
library("xgboost")
data(agaricus.train, package='xgboost')
data(agaricus.test, package='xgboost')
dtrain <- with(agaricus.train, xgboost::xgb.DMatrix(data, label = label))
dtest <- with(agaricus.test, xgboost::xgb.DMatrix(data, label = label))
watchlist <- list(train = dtrain, eval = dtest)
param <- list(max_depth = 2, eta = 1, nthread = 2,
                objective = "binary:logitraw", eval_metric = "auc")
weight <- read.table("weight.txt", header=FALSE)[,2]
summary(weight)
xgboost::setinfo(dtrain, 'weight', weight)
# nrounds=32 ok
bst <- xgboost::xgb.train(params=param, data=dtrain, nrounds = 32,              watchlist=watchlist, verbose=2)
# nrounds=33 crashed
bst <- xgboost::xgb.train(params=param, data=dtrain, nrounds = 33,              watchlist=watchlist, verbose=2)
>
[1] train-auc:0.962643  eval-auc:0.960373
 [2] train-auc:0.985768  eval-auc:0.979930
 [3] train-auc:0.998551  eval-auc:0.998518
 [4] train-auc:0.999061  eval-auc:0.998943
 [5] train-auc:0.999566  eval-auc:1.000000
 [6] train-auc:0.999566  eval-auc:1.000000
 [7] train-auc:0.999566  eval-auc:1.000000
 [8] train-auc:0.999561  eval-auc:1.000000
 [9] train-auc:0.999561  eval-auc:1.000000
 [10]    train-auc:0.999561  eval-auc:1.000000
 [11]    train-auc:0.999561  eval-auc:1.000000
 [12]    train-auc:0.999561  eval-auc:1.000000
 [13]    train-auc:0.999566  eval-auc:1.000000
 [14]    train-auc:0.999566  eval-auc:1.000000
 [15]    train-auc:0.999566  eval-auc:1.000000
 [16]    train-auc:0.999612  eval-auc:1.000000
[17]    train-auc:0.999719  eval-auc:1.000000
 [18]    train-auc:0.999780  eval-auc:1.000000
 [19]    train-auc:0.999892  eval-auc:1.000000
 [20]    train-auc:0.999935  eval-auc:1.000000
 [21]    train-auc:0.999948  eval-auc:1.000000
 [22]    train-auc:0.999972  eval-auc:1.000000
 [23]    train-auc:0.999974  eval-auc:1.000000
 [24]    train-auc:0.999986  eval-auc:1.000000
 [25]    train-auc:0.999990  eval-auc:1.000000
 [26]    train-auc:0.999994  eval-auc:1.000000
 [27]    train-auc:0.999998  eval-auc:1.000000
 [28]    train-auc:0.999998  eval-auc:1.000000
 [29]    train-auc:0.999999  eval-auc:1.000000
 [30]    train-auc:0.999999  eval-auc:1.000000
 [31]    train-auc:1.000000  eval-auc:1.000000
 [32]    train-auc:1.000000  eval-auc:1.000000
Error in xgb.iter.eval(bst$handle, watchlist, iteration - 1, feval) :
   [12:02:28] src/metric/auc.cc:322: Check failed: auc <= local_area (1.04939e+  07 vs. 1.04939e+07) :
 Stack trace:
   [bt] (0) /home/zhu/R/x86_64-pc-linux-gnu-library/4.1/xgboost/libs/xgboost.    so(+0x571ec) [0x7fe35d6571ec]
   [bt] (1) /home/zhu/R/x86_64-pc-linux-gnu-library/4.1/xgboost/libs/xgboost.    so(+0x8beb2) [0x7fe35d68beb2]
   [bt] (2) /home/zhu/R/x86_64-pc-linux-gnu-library/4.1/xgboost/libs/xgboost.    so(+0x22bda5) [0x7fe35d82bda5]
   [bt] (3) /home/zhu/R/x86_64-pc-linux-gnu-library/4.1/xgboost/libs/xgboost.    so(XGBoosterEvalOneIter+0x204) [0x7fe35d893cb4]
   [bt] (4) /home/zhu/R/x86_64-pc-linux-gnu-library/4.1/xgboost/libs/xgboost.    so(XGBoosterEvalOneIter_R+0x2d4) [0x7fe35d655ec4]
   [bt] (5) /usr/lib/R/lib/libR.so(+0xfa1a4) [0x7fe3652a71a4]
   [bt] (6) /usr/lib/R/lib/libR.so(+0x13b1c0) [0x7fe3652e81c0]
   [bt] (7) /usr/lib/R/lib/libR.so(Rf_eval+0x180) [0x7fe3652f2700]
   [bt] (8) /usr/lib/R/lib/libR.so(+0x14751f) [0x7fe3652f451f]

system information:
Ubuntu 18.04 
packageVersion("xgboost")
[1] ‘1.7.7.1’

**Comment by trivialfis** [comment]
Apologies for the slow reply. Appears to be running into a floating error. Will try to open a PR for squashing the output auc
